src/test/java/bio/terra/service/dataset/DatasetBucketDaoTest.java

```diff
@@ -0,0 +1,246 @@
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+@AutoConfigureMockMvc
+@Category(Unit.class)
+public class DatasetBucketDaoTest {
+
+    private static final Logger logger = LoggerFactory.getLogger(DatasetBucketDaoTest.class);
+
+    @Autowired
+    private JsonLoader jsonLoader;
+
+    @Autowired
+    private DatasetDao datasetDao;
+
+    @Autowired
+    private DatasetBucketDao datasetBucketDao;
+
+    @Autowired
+    private ProfileDao profileDao;
+
+    @Autowired
+    private GoogleResourceDao resourceDao;
+
+    @Autowired
+    private NamedParameterJdbcTemplate jdbcTemplate;
+
+    @Autowired
+    private ResourceService resourceService;
+
+    private BillingProfileModel billingProfile;
+    private GoogleBucketResource bucketForFile;
+    private UUID projectId;
+    private Dataset dataset;
+
+    private UUID datasetId;
+    private UUID bucketResourceId;
+
+    private UUID createDataset(DatasetRequestModel datasetRequest, String newName) throws Exception {
+        datasetRequest.name(newName).defaultProfileId(billingProfile.getId());
+        dataset = DatasetUtils.convertRequestWithGeneratedNames(datasetRequest);
+        dataset.projectResourceId(projectId);
+        String createFlightId = UUID.randomUUID().toString();
+        UUID datasetId = UUID.randomUUID();
+        dataset.id(datasetId);
+        datasetDao.createAndLock(dataset, createFlightId);
+        datasetDao.unlockExclusive(dataset.getId(), createFlightId);
+        return datasetId;
+    }
+
+    private UUID createBucket() throws InterruptedException {
+        String ingestFileFlightId = UUID.randomUUID().toString();
+        bucketForFile =
+            resourceService.getOrCreateBucketForFile(
+                dataset.getName(),
+                billingProfile,
+                ingestFileFlightId);
+        return bucketForFile.getResourceId();
+    }
+
+    private UUID createDataset(String datasetFile) throws Exception  {
+        DatasetRequestModel datasetRequest = jsonLoader.loadObject(datasetFile, DatasetRequestModel.class);
+        return createDataset(datasetRequest, datasetRequest.getName() + UUID.randomUUID().toString());
+    }
+
+    @Before
+    public void setup() {
+        logger.info("-------------------Setup----------------------");
```

Comment:
Just curious if we want to include these log statements since I don't think we do it anywhere else?

Reply:
Nah, it was just for testing for me. I think a few of these comments were made on an old version of the code.

Reply:
got it